homeassistant/components/wallbox/number.py

```diff
@@ -23,20 +28,28 @@ async def async_setup_entry(hass, config, async_add_entities):
     except InvalidAuth:
         pass
     else:
-        async_add_entities([WallboxNumber(coordinator, config)])
+        async_add_entities(
+            [
+                WallboxNumber(coordinator, config, description)
+                for ent in filtered_data
```

Comment:
We don't need to pre-filter the coordinator data. It's enough to check if there's a description for the data key. Use `dict.get` to look up the key in the number types dict.

Reply:
OK, wil change this in new commit